src/main/java/dev/cookiecode/maven/plugin/protobuf/AbstractProtocMojo.java

```diff
@@ -900,6 +919,41 @@ protected List<File> findProtoFilesInDirectories(final Iterable<File> directorie
         return protoFiles;
     }
 
+    /**
+     * Sorts the given list of .proto files lexicographically.
+     * <p>
+     * This results in an order like
+     * <ol>
+     * <li>/top1/a.proto</li>
+     * <li>/top1/sub1/a.proto</li>
+     * <li>/top1/sub1/b.proto</li>
+     * <li>/top1/sub1/bot1/d.proto</li>
+     * <li>/top1/sub1/bot2/c.proto</li>
+     * <li>/top1/sub2/x.proto</li>
+     * <li>/top2/z.proto</li>
+     * <li>/top2/sub1/y.proto</li>
+     * </ol>
+     *
+     * @param unsortedFiles the unsorted list of files
+     * @return the sorted list
+     */
+    protected List<File> sortProtoFiles(final List<File> unsortedFiles) {
+        return unsortedFiles.stream().sorted((aFirst, aSecond) -> {
+            final Path first = aFirst.getAbsoluteFile().toPath();
+            final Path firstParent = first.getParent();
+            final Path second = aSecond.getAbsoluteFile().toPath();
+            final Path secondParent = second.getParent();
+
+            // In case of equal parent directories, compare the file names
+            if(firstParent.equals(secondParent)) {
+                return first.getFileName().compareTo(second.getFileName());
+            } else {
+                // For different parent directories, sort parents lexicographically
+                return firstParent.compareTo(secondParent);
+            }
+        }).collect(Collectors.toList());
+    }
+
```

Comment:
Thank you for your contribution, could you please write unit tests for this feature ?

(I discovered that this forked project do not have tests so that's definitively something I will have to work on)

If you write the tests for this feature I can validate it, otherwise it will require a bit more time before it can be integrated. (Wait that I setup the test deps etc)

Reply:
Great to see you investing time into the maintenance of this fork!

I have now updated my pull request and added a test (not exactly a unit test cause the project does not seem to have unit testing infrastructure, but an "IT test", whatever that means, maybe integration test?) which validates that .proto files are supplied to the compiler in the correct and expected order if the new feature is enabled.

Reply:
Thank you for your words and also your efforts for this tests it's very appreciated.

Exact no unit tests at this stage but definitely something I want to add in the future 👍

Will do my best to release soon with your feature 🖖